src/main/scala/mesosphere/marathon/core/CoreGuiceModule.scala

```diff
@@ -238,4 +238,7 @@ class CoreGuiceModule(config: Config) extends AbstractModule {
 
   @Provides @Singleton
   def scheduler(coreModule: CoreModule): Scheduler = coreModule.mesosHeartbeatMonitor
+
+  @Provides @Singleton
+  def heartbeatMonitor(coreModule: CoreModule) = coreModule.mesosHeartbeatMonitor
```

Comment:
Not proud of this part since both `scheduler` and `heartbeatMonitor` are the same object, but injecting a `HeartbeatMonitor` where Mesos `Scheduler` is needed seems worse.